### DISCUSSION TITLE
Error ring information for large system

### DISCUSSION THREAD

**Discussion by zhao57** [rdkit/rdkit#4227] [Q&A]
Dear All,
I use the simple code below to achiveve the ring structure as follows,

from rdkit import Chem
m = Chem.MolFromMolFile('222.mol')
ssr = Chem.GetSymmSSSR(m)
num_ring = len(ssr)
print("num of ring",num_ring)
for ring in ssr:
    print("ring consisted of atoms id:",list(ring))

for small system (atoms may around 500), it works well, however when the system is around 600 atoms or more.
Does it has any method do ring detection for large system.

Best wishes!
zhao

**Comment by greglandrum** [comment]
Hi @zhao57
You didn't explain what problem you're encountering here.
If you want us to try and help with a specific problem, you will also need to share the input file which causes the problem.

**Comment by zhao57** [comment]
Dear Greg,
Thanks for your suggestions. The error information for large system is
shown as follows,

[20:19:29] non-ring atom 260 marked aromatic
Traceback (most recent call last):
  File "test.py", line 3, in <module>
    ssr = Chem.GetSymmSSSR(m)
Boost.Python.ArgumentError: Python argument types in
    rdkit.Chem.rdmolops.GetSymmSSSR(NoneType)
did not match C++ signature:
    GetSymmSSSR(RDKit::ROMol {lvalue})

The attached files are the input files. name 222.mol can run successfully,
However, name 456.mol meets the problem.

Best wishes!

zhao

************************************************************************
Yinbo Zhao, Ph.D.
Research Asisstant Professor
Department of Aerospace Engineering, Tohoku University
6-6-11-1302-1, Aza-Aoba, Aramaki, Aobaku; Sendai 980-8579, Japan/
Institute of Fluid Science, Tohoku University
2-1-1 Katahira, Aoba-ku, Sendai, 980-8577, Japan
E-mail: ***@***.***
************************************************************************


Greg Landrum ***@***.***> 于2021年6月9日周三 下午10:28写道：

**Reply by greglandrum** [reply]
This doesn't look like a ring finding problem.
If you look at the output you will see that there was a problem processing the large system:
```
[20:19:29] non-ring atom 260 marked aromatic
```
There's an error somewhere in that input file that would need to be fixed.

**Reply by zhao57** [reply]
Dear Greg,
I tried and just added the "sanitize=False" and it works.
However,actually i did not understand the reason. If it is avoided the valence check and it works. But the type of monomers in the system is the same no matter for the small one or large one.

Best wishes!

zhao
